### PR TITLE
[MRG] Update CHANGELOG for the Gensim 4.1.0 release

### LINKED ISSUE #3199: Review change log for 4.1.0
Gentlemen, I'm planning to do the next release in the immediate future.

One thing I'd like to do differently for this release is to review the change log _before_ we do the release. Previously, we updated the change log as part of the release project, which made review of the change log itself impossible, because by the time the changes were visible, the new version was already on PyPI and it was too late to make changes.

This time around, I've kept the change log up to date as we merged each PR. This has two benefits:

1. Anyone can spot errors in the change log and correct them.
2. Anyone can see how many of each issue type (bug fix, documentation, improvement, etc) we've made since the last release. Previously, this information was only visible via the issue tracker, and only a select few bother to dig around there.

Can you please have a look at the change log and make any required changes? Things like wording, omission of less relevant PRs, wrong PR category, missed PR, etc.

For the last part, here is the list of merged PRs since the last release: https://github.com/RaRe-Technologies/gensim/pulls?q=is%3Apr+is%3Aclosed+closed%3A%3E2021-04-01

### PR DESCRIPTION
**Pull request by piskvorky**
Fixes #3199.

@mpenkov I also linked from CHANGELOG to the new EnsambleLDA tutorial which isn't "live" yet: https://radimrehurek.com/gensim/auto_examples/tutorials/run_ensemblelda.html

Once we release & rebuild & upload the docs, the link should work. But let's verify.

### CODE REVIEW

**Comment by mpenkov** [comment]
@piskvorky I added additional info about backwards incompatibilities and fixed some typos, please have a look.

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,16 +3,69 @@ Changes
...
+model.infer_vector(..., epochs=123)
+```
+
+instead of this:
+
+```python
+model.infer_vector(..., steps=123)
+```
+
+### Hidden KEY_TYPES in gensim/models/keyedvectors.py
```
Comment (piskvorky) [inline CHANGELOG.md:60]: I'd propose removing this. Completely internal, irrelevant to everyone, IMO.

**Review by piskvorky — COMMENTED** [review]
Thanks for the fixes, looks good!

File: CHANGELOG.md (hunk 1 above)
Comment (mpenkov) [reply CHANGELOG.md:60]: OK.

**Comment by piskvorky** [comment]
FYI, running the `generate_changelog.py` script failed with some 403 API error:

I contacted the Github support, this is their answer:

> Thanks for writing in to GitHub Support!

> I was able to find in our logs some of the instances of the request made to that endpoint( https://api.github.com/repos/RaRe-Technologies/gensim...).

> You're hitting the abuse rate limits and are different from our primary rate limits which are documented here:

> https://docs.github.com/free-pro-team@latest/rest/overview/resources-in-the-rest-api#rate-limiting

> The abuse rate limits are primarily here to ensure the API's availability by limiting spikes in requests, especially expensive requests. This normally happens when users make lots of requests in a short period of time, especially requests that consume a lot of resources, and also when making requests concurrently.

> There's documentation on the abuse rate limits here along with advice on avoiding them and what to do if you do hit them:

> https://docs.github.com/free-pro-team@latest/rest/guides/best-practices-for-integrators#dealing-with-abuse-rate-limits

> The abuse rate limits are necessary and ensure the stability of the platform for all customers; we're not able to disable or increase them. We recommend checking out that guide to make sure you're able to handle these responses and let us know if you have any other questions about this!

**Review by mpenkov — APPROVED** [review]

**Pull request merged**